src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -244,167 +96,17 @@ public class DeploymentOptions
-            _displayNameMapDict[nameof(d.IgnoreWhitespace)] = "Ignore Whitespace";
-            _displayNameMapDict[nameof(d.IgnoreWithNocheckOnCheckConstraints)] = "Ignore With Nocheck On Check Constraints";
-            _displayNameMapDict[nameof(d.IgnoreWithNocheckOnForeignKeys)] = "Ignore With Nocheck On Foreign Keys";
-            _displayNameMapDict[nameof(d.IgnoreWorkloadClassifiers)] = "Ignore Workload Classifiers";
-            _displayNameMapDict[nameof(d.IncludeCompositeObjects)] = "Include Composite Objects";
-            _displayNameMapDict[nameof(d.IncludeTransactionalScripts)] = "Include Transactional Scripts";
-            _displayNameMapDict[nameof(d.IsAlwaysEncryptedParameterizationEnabled)] = "Is Always Encrypted Parameterization Enabled";
-            _displayNameMapDict[nameof(d.LongRunningCommandTimeout)] = "Long Running Command Timeout";
-            _displayNameMapDict[nameof(d.NoAlterStatementsToChangeClrTypes)] = "No Alter Statements To Change Clr Types";
-            _displayNameMapDict[nameof(d.PopulateFilesOnFileGroups)] = "Populate Files On File Groups";
-            _displayNameMapDict[nameof(d.PreserveIdentityLastValues)] = "Preserve Identity Last Values";
-            _displayNameMapDict[nameof(d.RebuildIndexesOfflineForDataPhase)] = "Rebuild Indexes Offline For Data Phase";
-            _displayNameMapDict[nameof(d.RegisterDataTierApplication)] = "Register Data Tier Application";
-            _displayNameMapDict[nameof(d.RestoreSequenceCurrentValue)] = "Restore Sequence Current Value";
-            _displayNameMapDict[nameof(d.RunDeploymentPlanExecutors)] = "Run Deployment Plan Executors";
-            _displayNameMapDict[nameof(d.ScriptDatabaseCollation)] = "Script Database Collation";
-            _displayNameMapDict[nameof(d.ScriptDatabaseCompatibility)] = "Script Database Compatibility";
-            _displayNameMapDict[nameof(d.ScriptDatabaseOptions)] = "Script Database Options";
-            _displayNameMapDict[nameof(d.ScriptDeployStateChecks)] = "Script Deploy State Checks";
-            _displayNameMapDict[nameof(d.ScriptFileSize)] = "Script File Size";
-            _displayNameMapDict[nameof(d.ScriptNewConstraintValidation)] = "Script New Constraint Validation";
-            _displayNameMapDict[nameof(d.ScriptRefreshModule)] = "Script Refresh Module";
-            _displayNameMapDict[nameof(d.SqlCommandVariableValues)] = "Sql Command Variable Values";
-            _displayNameMapDict[nameof(d.TreatVerificationErrorsAsWarnings)] = "Treat Verification Errors As Warnings";
-            _displayNameMapDict[nameof(d.UnmodifiableObjectWarnings)] = "Unmodifiable Object Warnings";
-            _displayNameMapDict[nameof(d.VerifyCollationCompatibility)] = "Verify Collation Compatibility";
-            _displayNameMapDict[nameof(d.VerifyDeployment)] = "Verify Deployment";
-            #endregion
-        }
+        public Dictionary<string, DeploymentOptionProperty<bool>> BooleanOptionsDict { get; set; }
```

Comment:
nit: update the name to be "BooleanOptionsDictionary" so it's clear that it's a dictionary. No need to abbreviate it